daemon/workit.cpp

```diff
@@ -501,11 +522,13 @@ int work_it(CompileJob &j, unsigned int job_stat[], MsgChannel *client, CompileR
                     }
 
                     kill(pid, SIGTERM); // Most likely crashed anyway ...
-                    return_value = EXIT_COMPILER_CRASHED;
-                    client_fd = -1;
-                    input_complete = true;
+                    if (input_complete) {
+                        return_value = EXIT_COMPILER_CRASHED;
+                    }
                     delete fcmsg;
                     fcmsg = 0;
+                    close(sock_in[1]);
+                    sock_in[1] = -1;
```

Comment:
Generally speaking, I agree with @HenryMiller1 and I prefer to use constants instead of magic numbers like -1.  However, I chose not to use a constant in this commit because the existing code was using -1 and I thought maintaining that consistency was important.  I also thought that cleaning up the existing code to use a named constant was outside the scope of this bug fix and should be done in a separate PR, if at all.

May I suggest using -1 for this commit for now?  You two can discuss and do a full cleanup of the existing code in a separate PR if a named constant is preferred over -1.